### DISCUSSION TITLE
Overwriting .cirrus.yml

### DISCUSSION THREAD

**Discussion by 5u623l20** [cirruslabs/cirrus-ci-docs#1003] [Ideas]
I am looking for a situation when an upstream has already a .cirrus.yml file in the repo and my tests overwrite it. This creates a merge conflict in between. So is there any way like there is a directory of .cirrus-ci and we could add multiple files like the way we can do in .github/workflows?

**Comment by fkorotkov** [comment]
There is no similar to `.github/workflows` option, you can only define tasks in `.cirrus.yml` and `.cirrus.star` files. The latter has more flexibility like [external module loading](https://cirrus-ci.org/guide/programming-tasks/#module-loading) but no option to load everything in a directory.

If you upstream has only `.cirrus.yml` then a workaround might be to define your new tasks in `.cirrus.star`.

**Reply by 5u623l20** [reply]
Which one takes precedence if we have both?

**Reply by 5u623l20** [reply]
Found the answer although it's not satisfactory. :'(
"In regard to top-level overrides, note that when using both YAML and Starlark configuration formats they get merged and the YAML configuration always comes first."

**Comment by 5u623l20** [comment]
Let's say we have both a `.cirrus.yml` and `.cirrus.star` is there any tools we can use to find the generated final `.cirrus.yml` before wasting CPU cycles.

**Reply by 5u623l20** [reply]
You should put this "https://github.com/cirruslabs/cirrus-cli/blob/master/STARLARK-MODULES.md#testing" in the original docs rather than in the modules page. This is a real lifesaver. :D
